src/sync/sync_manager.cpp

```diff
@@ -303,6 +303,26 @@ std::vector<std::shared_ptr<SyncUser>> SyncManager::all_logged_in_users() const
     return users;
 }
 
+std::shared_ptr<SyncUser> SyncManager::get_current_user() const
+{
+    std::lock_guard<std::mutex> lock(m_user_mutex);
+    
+    std::shared_ptr<SyncUser> result = nullptr;
```

Comment:
I'd be inclined to write this as something like:
```c++
auto is_active_user = [](auto& el) { return el.second->state() == SyncUser::State::Active; };
auto it = std::find_if(m_users.begin(), m_users.end(), is_active_user);
if (it == m_users.end())
    return nullptr;

if (std::find_if(it, m_users.end(), is_active_user) != m_users.end())
    throw std::logic_error("get_current_user cannot be called if more that one valid, logged-in user exists.");

return it->second;
```

That better highlights what the function is doing.